src/server/api/routers/scoreboard.ts

```diff
@@ -0,0 +1,102 @@
+// types.ts
+type Challenge = {
+  roundId: number;
+  points: number;
+  teamId: string;
+};
+
+type TeamScores = {
+  teamId: string;
+  teamName: string;
+  rounds: Record<
+    number,
+    {
+      challengeA: number;
+      challengeB: number;
+      challengeC: number;
+    }
+  >;
+  total: number;
+};
+
+// server/api/routers/scores.ts
+import { z } from "zod";
+import { createTRPCRouter, publicProcedure } from "../trpc";
+
+export const scoreboardRouter = createTRPCRouter({
+  getScoreboard: publicProcedure.query(async ({ ctx }) => {
+    // Fetch all teams
+    const teams = await ctx.db.team.findMany({
+      select: {
+        id: true,
+        name: true,
+      },
+    });
+
+    // Fetch all challenges
+    const [challengesA, challengesB, challengesC] = await Promise.all([
+      ctx.db.challengeA.findMany(),
+      ctx.db.challengeB.findMany(),
+      ctx.db.challengeC.findMany(),
+    ]);
+
+    // Process scores for each team
+    const teamScores: TeamScores[] = teams.map((team) => {
```

Comment:
Hmm, I think this way of computing the scores could be improved.
Currently, TC -> O(t * (cA + cB + cC)) where t-> number of teams, and the other variables represent the count of challenges.

If you have some spare time, consider modifying this.

Maybe have a dict where each key represents a team and points to the scores. That way this could be computed in O(cA + cB + cC)